### PR TITLE
jewel: rbd: rbd-nbd: kernel reported invalid device size (0, expected 1073741824)

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/20017

### CODE REVIEW

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 1]
```diff
@@ -469,6 +469,10 @@ static int open_device(const char* path, bool try_load_moudle = false)
 
 static int check_device_size(int nbd_index, unsigned long expected_size)
 {
+  // There are bugs with some older kernel versions that result in an
```
Comment (amitkumar50) [inline src/tools/rbd_nbd/rbd-nbd.cc]: @smithfarm I believe multi-line comment should be inside /* */

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 2]
```diff
@@ -480,6 +484,12 @@ static int check_device_size(int nbd_index, unsigned long expected_size)
   ifs >> size;
   size *= RBD_NBD_BLKSIZE;
 
+  if (size == 0) {
+    // Newer kernel versions will report real size only after nbd
```
Comment (amitkumar50) [inline src/tools/rbd_nbd/rbd-nbd.cc]: @smithfarm I believe multi-line comment should be inside /* */

**Comment by smithfarm** [comment]
@amitkumar50 Backports are cherry-picked from master, so any changes would have to go into master, first.

**Comment by smithfarm** [comment]
@dillaman @trociny This PR was included in this run, which is exhibiting a new failure: http://pulpito.ceph.com/smithfarm-2017-08-24_15:58:59-rbd-wip-jewel-backports-distro-basic-smithi/

**Comment by smithfarm** [comment]
@dillaman This passed an rbd suite at http://tracker.ceph.com/issues/20613#note-22 and a rados suite at http://tracker.ceph.com/issues/20613#note-20

Please review.

**Pull request merged**